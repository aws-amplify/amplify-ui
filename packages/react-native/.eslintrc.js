module.exports = {
  ignorePatterns: ['dist', '.eslintrc.js', '*.config.js', 'jest.setup.js'],
  extends: ['@aws-amplify/amplify-ui/react'],
  overrides: [
    {
      extends: ['@aws-amplify/amplify-ui/jest'],
      files: ['**/__mocks__/**', '**/__tests__/**'],
    },
  ],
  // point to local tsconfig
  parserOptions: { project: ['tsconfig.json'], tsconfigRootDir: __dirname },
};
