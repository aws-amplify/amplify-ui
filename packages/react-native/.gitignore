dist
.watchmanconfig

# ignore jest generated coverage directory
coverage 
