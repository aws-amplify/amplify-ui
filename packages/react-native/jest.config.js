module.exports = {
  preset: 'react-native',
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  collectCoverage: true,
  collectCoverageFrom: [
    '<rootDir>/src/**/*.{js,jsx,ts,tsx}',
    '!<rootDir>/src/**/*{c,C}onstants.ts',
  ],
  moduleNameMapper: {
    '^react$': '<rootDir>/node_modules/react',
    '^react-native$': '<rootDir>/node_modules/react-native',
  },
  coverageThreshold: {
    global: {
      branches: 90,
      functions: 90,
      lines: 90,
      statements: 90,
    },
  },
  setupFiles: ['<rootDir>/jest.setup.js'],
};
