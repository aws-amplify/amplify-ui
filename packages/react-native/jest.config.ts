import { Config } from 'jest';

const config: Config = {
  preset: 'react-native',
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  collectCoverage: true,
  collectCoverageFrom: [
    '<rootDir>/src/**/*.{js,jsx,ts,tsx}',
    '!<rootDir>/src/**/*{c,C}onstants.ts',
  ],
  moduleNameMapper: {
    '^react-native$': '<rootDir>/node_modules/react-native',
    '^uuid$': '<rootDir>/../../node_modules/uuid',
  },
  modulePaths: ['<rootDir>/node_modules/'],
  coverageThreshold: {
    global: {
      branches: 90,
      functions: 90,
      lines: 90,
      statements: 90,
    },
  },
  setupFiles: ['<rootDir>/jest.setup.ts'],
};

export default config;
