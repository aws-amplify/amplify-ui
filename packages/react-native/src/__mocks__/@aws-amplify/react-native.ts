export default {
  loadUrlPolyfill: jest.fn(),
};
