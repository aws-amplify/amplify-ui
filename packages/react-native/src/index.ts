export {
  Authenticator,
  AuthenticatorProps,
  useAuthenticator,
  UseAuthenticator,
  withAuthenticator,
  WithAuthenticatorOptions,
} from './Authenticator';
export * from './InAppMessaging';
export {
  Theme,
  ThemeProvider,
  defaultDarkModeOverride,
  defaultTheme,
  useTheme,
} from './theme';
