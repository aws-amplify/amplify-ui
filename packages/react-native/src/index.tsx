import React from 'react';
import { Text, View } from 'react-native';

export function HelloWorld(): JSX.Element {
  return (
    <View>
      <Text>Hello World!</Text>
    </View>
  );
}
