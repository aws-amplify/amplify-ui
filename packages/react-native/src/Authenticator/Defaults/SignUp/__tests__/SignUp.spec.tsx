import React from 'react';
import { render } from '@testing-library/react-native';

import { SignUp } from '..';
import { authenticatorTextUtil } from '@aws-amplify/ui';

const USERNAME = 'username';
const username = {
  name: USERNAME,
  label: 'Username',
  placeholder: 'Username',
  type: 'default' as const,
};
const password = {
  name: 'password',
  label: 'Password',
  placeholder: 'Password',
  type: 'password' as const,
};
const confirmPassword = {
  name: 'confirmPassword',
  label: 'Confirm Password',
  placeholder: 'Confirm Password',
  type: 'password' as const,
};
const phone = {
  name: 'phone',
  label: 'Phone',
  placeholder: 'Phone',
  type: 'phone' as const,
};

const fields = [username, password, confirmPassword, phone];

const props = {
  fields,
  FormFields: SignUp.FormFields,
  Footer: SignUp.Footer,
  handleBlur: jest.fn(),
  handleChange: jest.fn(),
  handleSubmit: jest.fn(),
  hasValidationErrors: false,
  Header: SignUp.Header,
  isPending: false,
  socialProviders: [],
  toResetPassword: jest.fn(),
  toFederatedSignIn: jest.fn(),
  toSignIn: jest.fn(),
  validationErrors: undefined,
};

const { getCreatingAccountText } = authenticatorTextUtil;

describe('SignUp', () => {
  it('renders as expected', () => {
    const { toJSON, getAllByRole, queryByText } = render(<SignUp {...props} />);
    expect(toJSON()).toMatchSnapshot();

    expect(queryByText(getCreatingAccountText())).toBe(null);
    expect(getAllByRole('text')).toHaveLength(fields.length);
  });

  it('renders as expected with errors', () => {
    const error = 'Something went wrong';
    const { toJSON, getByRole, queryByText } = render(
      <SignUp {...props} error={error} />
    );
    expect(toJSON()).toMatchSnapshot();

    expect(getByRole('alert')).toBeDefined();
    expect(queryByText(error)).toBeDefined();
  });

  it('renders as expected when hideSignIn is true', () => {
    const { toJSON, queryByRole } = render(<SignUp {...props} hideSignIn />);
    expect(toJSON()).toMatchSnapshot();

    expect(queryByRole('tab')).toBe(null);
  });

  it('renders as expected when isPending is true', () => {
    const { toJSON, queryByText } = render(<SignUp {...props} isPending />);
    expect(toJSON()).toMatchSnapshot();

    expect(queryByText(getCreatingAccountText())).toBeDefined();
  });

  it('renders as expected with validationErrors', () => {
    const { toJSON, getByText } = render(
      <SignUp
        {...props}
        isPending
        validationErrors={{ [USERNAME]: ['error', 'another error'] }}
      />
    );

    expect(toJSON()).toMatchSnapshot();

    expect(getByText('error')).toBeDefined();
    expect(getByText('another error')).toBeDefined();
  });
});
