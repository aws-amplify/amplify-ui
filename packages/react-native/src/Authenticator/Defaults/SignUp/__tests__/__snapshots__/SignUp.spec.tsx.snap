// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders as expected 1`] = `"SignUp"`;
